From c3174b0905ce910f4905f7991c79c355d4ac1754 Mon Sep 17 00:00:00 2001
From: Michael Achenbach <machenbach@chromium.org>
Date: Mon, 7 Dec 2020 13:00:22 +0100
Subject: [PATCH] [MB][test] Add builders for RISC-V

No-Try: true
Bug: v8:11233
Change-Id: I391bbd22b4415e9a669795b62b4933fb4b3e9eef
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/2575124
Reviewed-by: Georg Neis <neis@chromium.org>
Commit-Queue: Michael Achenbach <machenbach@chromium.org>
Auto-Submit: Michael Achenbach <machenbach@chromium.org>
Cr-Commit-Position: refs/heads/master@{#71643}
---
 infra/mb/mb_config.pyl     |  9 +++++++++
 infra/testing/builders.pyl | 23 +++++++++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/infra/mb/mb_config.pyl b/infra/mb/mb_config.pyl
index 245ba4b0fcc..c540f803c9f 100644
--- a/infra/mb/mb_config.pyl
+++ b/infra/mb/mb_config.pyl
@@ -169,6 +169,8 @@
       # IBM.
       'V8 Linux - ppc64 - sim': 'release_simulate_ppc64',
       'V8 Linux - s390x - sim': 'release_simulate_s390x',
+      # RISC-V
+      'V8 Linux - riscv64 - sim - builder': 'release_simulate_riscv64',
     },
     'client.v8.branches': {
       'V8 Linux - previous branch': 'release_x86',
@@ -237,6 +239,7 @@
       'v8_linux64_cfi_rel_ng': 'release_x64_cfi',
       'v8_linux64_fuzzilli_ng': 'release_x64_fuzzilli',
       'v8_linux64_msan_rel_ng': 'release_simulate_arm64_msan_minimal_symbols',
+      'v8_linux64_riscv64_rel_ng': 'release_simulate_riscv64',
       'v8_linux64_tsan_rel_ng': 'release_x64_tsan_minimal_symbols',
       'v8_linux64_tsan_no_cm_rel_ng': 'release_x64_tsan_no_cm',
       'v8_linux64_tsan_isolates_rel_ng':
@@ -401,6 +404,8 @@
       'release_bot', 'simulate_mips64el'],
     'release_simulate_ppc64': [
       'release_bot', 'simulate_ppc64'],
+    'release_simulate_riscv64': [
+      'release_bot', 'simulate_riscv64'],
     'release_simulate_s390x': [
       'release_bot', 'simulate_s390x'],
 
@@ -783,6 +788,10 @@
       'gn_args': 'target_cpu="x64" v8_target_cpu="ppc64"',
     },
 
+    'simulate_riscv64': {
+      'gn_args': 'target_cpu="x64" v8_target_cpu="riscv64"',
+    },
+
     'simulate_s390x': {
       'gn_args': 'target_cpu="x64" v8_target_cpu="s390x"',
     },
diff --git a/infra/testing/builders.pyl b/infra/testing/builders.pyl
index bff428403f4..426b0bdc606 100644
--- a/infra/testing/builders.pyl
+++ b/infra/testing/builders.pyl
@@ -549,6 +549,16 @@
     ],
   },
   ##############################################################################
+  # Linux64 with RISC-V simulators
+  'v8_linux64_riscv64_rel_ng_triggered': {
+    'swarming_dimensions': {
+      'os': 'Ubuntu-16.04',
+    },
+    'tests': [
+      {'name': 'v8testing', 'shards': 3},
+    ],
+  },
+  ##############################################################################
   # Odroids with native arm
   'v8_odroid_arm_rel_ng_triggered': {
     'swarming_dimensions' : {
@@ -1741,6 +1751,19 @@
       {'name': 'v8testing', 'shards': 3},
     ],
   },
+  'V8 Linux - riscv64 - sim': {
+    'swarming_dimensions': {
+      'os': 'Ubuntu-16.04',
+    },
+    'swarming_task_attrs': {
+      'expiration': 14400,
+      'hard_timeout': 3600,
+      'priority': 35,
+    },
+    'tests': [
+      {'name': 'v8testing', 'shards': 3},
+    ],
+  },
   'V8 Linux - s390x - sim': {
     'swarming_dimensions': {
       'os': 'Ubuntu-16.04',
-- 
2.35.1

